pkg/network/java/testutil/JavaClientSimulator/src/main/java/Clients/JavaClients.java

```diff
@@ -19,118 +21,149 @@
-                Response response = call.execute();
-                System.out.println("response: " + response);
-                Thread.sleep(1000);
-            }
-        } else{
-            while (iterations > 0) {
-                OkHttpClient client = new OkHttpClient();
-                Call call = client.newCall(request);
-                Response response = call.execute();
-                System.out.println("response: " + response);
-                iterations--;
-            }
+    private static ClientType getClientType(String clientTypeArg) throws ParseException {
+        try {
+            return ClientType.valueOf(clientTypeArg.toLowerCase());
+        } catch (IllegalArgumentException e) {
+            return ClientType.unsupported;
         }
     }
 
-    public static void HTTPApacheClientExample(int iterations, String url) throws InterruptedException, IOException {
-        if (iterations == -1){
-            while (true){
-                CloseableHttpClient httpClient = HttpClients.createDefault();
-                try {
-                    HttpGet request = new HttpGet("https://"+url);
-                    CloseableHttpResponse response = httpClient.execute(request);
-                    System.out.println("response: " + response);
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-                finally {
-                    httpClient.close();
-                    Thread.sleep(1000);
-                }
+    public static void executeCallback(String clientTypeArg, int iterations, long sleepInterval, String url) throws InterruptedException, ParseException {
 
-            }
-        } else{
-            while (iterations > 0) {
-                CloseableHttpClient httpClient = HttpClients.createDefault();
-                try {
-                    HttpGet request = new HttpGet("https://"+url);
-                    CloseableHttpResponse response = httpClient.execute(request);
-                    System.out.println("response: " + response);
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-                finally {
-                    httpClient.close();
-                    iterations--;
-                }
-            }
+        ClientType clientType = getClientType(clientTypeArg);
+
+        System.out.println("URL: " + url);
+        System.out.println("Iterations: " + iterations);
+        System.out.println("Interval: " + sleepInterval);
+
+        Runnable callback = null;
+
+        // Execute handler based on client type
+        switch (clientType) {
+            case apache:
+                System.out.println("Executing handler for Apache Http client:");
+                callback = () -> {
+                    try {
+                        JavaClients.HTTPApacheClientExample(url);
+                    } catch (IOException e) {
+
+                    }
+                };
```

Comment:
```suggestion
                callback = () -> {
                    JavaClients.HTTPApacheClientExample(url);
                };
```

Reply:
a suggestion along with https://github.com/DataDog/datadog-agent/pull/17355/files/da4be5deac22480cf25155def2459f065f8b3613..39157c6bcb2c62be4c074a6a2853314582b952cb#r1219062520
as the current implementation basically copy pastes from one to another

Reply:
that won't work, the function throws IOException and must be wrapped